packages/discovery-provider/src/utils/auth_middleware.py

```diff
@@ -1,21 +1,57 @@
 import functools
 import logging
+from enum import Enum
+from typing import cast
 
 from eth_account.messages import encode_defunct
 from flask.globals import request
-from flask_restx import reqparse
+from flask_restx import Namespace, reqparse
 
 from src.models.users.user import User
+from src.queries.get_managed_users import (
+    GetUserManagersArgs,
+    get_user_managers_with_grants,
+)
 from src.utils import db_session, web3_provider
+from src.utils.helpers import decode_string_id
 
 logger = logging.getLogger(__name__)
 
 MESSAGE_HEADER = "Encoded-Data-Message"
 SIGNATURE_HEADER = "Encoded-Data-Signature"
 
 
+def is_active_manager(user_id: int, manager_id: int) -> bool:
+    try:
+        grants = get_user_managers_with_grants(
+            GetUserManagersArgs(user_id=user_id, is_approved=True, is_revoked=False)
+        )
+        for grant in grants:
+            manager = grant.get("manager")
+            if manager and manager.get("user_id") == manager_id:
+                return True
+    except Exception as e:
+        logger.error(
+            f"auth_middleware.py | Unexpected exception checking managers for user: {e}"
+        )
+    return False
+
+
+class AccessLevel(Enum):
+    ALL = "all"
```

Comment:
can we get some comments on what these mean?

Reply:
+1 I'm a little lost on these